source/common/upstream/maglev_lb.cc

```diff
@@ -57,18 +56,15 @@ MaglevTable::MaglevTable(const HostsPerLocality& hosts_per_locality,
   while (true) {
     for (uint64_t i = 0; i < table_build_entries.size(); i++) {
       TableBuildEntry& entry = table_build_entries[i];
-      // Only consider weight if we are doing weighted Maglev.
-      if (max_host_weight > 0) {
-        // Counts are in units of max_host_weight. To understand how counts_ and
-        // weight_ are used below, consider a host with weight equal to
-        // max_host_weight. This would be picked on every single iteration. If
-        // it had weight equal to backend_weight_scale / 3, then this would only
-        // happen every 3 iterations, etc.
-        if (iteration * entry.weight_ < entry.counts_) {
-          continue;
-        }
-        entry.counts_ += max_host_weight;
+      // Counts are in units of max_host_weight. To understand how counts_ and
+      // weight_ are used below, consider a host with weight equal to
+      // max_host_weight. This would be picked on every single iteration. If
+      // it had weight equal to backend_weight_scale / 3, then this would only
+      // happen every 3 iterations, etc.
+      if (iteration * entry.weight_ < entry.counts_) {
+        continue;
```

Comment:
Can you add a `ASSERT(max_host_weights) > 1);` here for sanity. Ideally we don't do anything different than the unweighted case that existed before when not weighting.

Reply:
Was this one incorrect? I'm curious if there's a simple `ASSERT` to indicate we are doing the unadorned construction for unweighted.

Reply:
Oh, I think I understand now... You want to skip this bit as an optimization if all weights are equal, is that right? To do that correctly, I think I'd actually need to track both the maximum and minimum effective weights (or something like that) to see if some weights are different.

Reply:
I don't want to necessarily skip it, I think we could add some `ASSERT`s to indicate it actually is doing the simple thing when we have all `1`s.

Reply:
got it